extension/src/fileSystem/watcher.ts

```diff
@@ -18,7 +19,7 @@ export const getRelativePattern = (
     if (isSameOrChild(workspaceFolderPath, path)) {
       return new RelativePattern(
         workspaceFolder,
-        join(relative(workspaceFolderPath, path), pattern)
+        joinWithForwardSlashes([relative(workspaceFolderPath, path), pattern])
```

Comment:
[Q] Can we create a new relativePattern file and move `joinWithForwardSlashes` and these function in there? Grouping the behaviour/functions together would give more clues about why it was done in the future.  

[I] Would be good to get a regression test for the behaviour (maybe add to `'should watch the .git directory for updates'` in `extension/src/test/suite/experiments/data/index.test.ts`).

Reply:
> [Q] Can we create a new relativePattern file and move joinWithForwardSlashes and these function in there? Grouping the behaviour/functions together would give more clues about why it was done in the future.

Done!

> [I] Would be good to get a regression test for the behaviour (maybe add to 'should watch the .git directory for updates' in extension/src/test/suite/experiments/data/index.test.ts).

Currently, we have a `.git` file watcher test set up for when the `.git` folder is outside of the workspace since that's how our `demo` project works. But I couldn't think of an uncomplicated way to test a workspace that has a local `.git` folder inside since we don't have a project like that in our repo... Is there something I'm misunderstanding/missing?

Reply:
not sure but can we create a set of nested subfolders inside of the demo project and then stub `getGitRepositoryRoot` to return that path or even stub `getGitPath`? Could then remove the top directory at the end of the test

Reply:
> not sure but can we create a set of nested subfolders inside of the demo project and then stub getGitRepositoryRoot to return that path or even stub getGitPath? Could then remove the top directory at the end of the test

Will work on it in a followup!